client/lib/cart-values/cart-items.js

```diff
@@ -1183,7 +1182,9 @@ export function isNextDomainFree( cart, domain = '' ) {
 export function isDomainBundledWithPlan( cart, domain ) {
 	const bundledDomain = get( cart, 'bundled_domain', '' );
 
-	return '' !== bundledDomain && toLower( domain ) === toLower( get( cart, 'bundled_domain', '' ) );
+	return (
+		'' !== bundledDomain && domain.toLowerCase() === get( cart, 'bundled_domain', '' ).toLowerCase()
```

Comment:
Theoretically, `domain` is a parameter and there's no guarantee it'll be provided. It doesn't make much sense for that to happen in this case, so I'll leave it to you to decide whether or not some bulletproofing is a good idea:

```suggestion
		'' !== bundledDomain && ( domain ?? '' ).toLowerCase() === get( cart, 'bundled_domain', '' ).toLowerCase()
```

Reply:
I agree, we can't guarantee it won't be nullish. Updated in 0802199ccf3169e9e1aca0403a0ced2a2f9d3ad5